src/test/java/teammates/storage/api/NotificationsDbTest.java

```diff
@@ -0,0 +1,272 @@
+import teammates.common.exception.InvalidParametersException;
+import teammates.test.BaseTestCaseWithLocalDatabaseAccess;
+
+/**
+ * SUT: {@link NotificationsDb}.
+ */
+public class NotificationsDbTest extends BaseTestCaseWithLocalDatabaseAccess {
+
+    private final NotificationsDb notificationsDb = NotificationsDb.inst();
+    private final Map<String, NotificationAttributes> typicalNotifications = getTypicalDataBundle().notifications;
+
+    // Notification attributes will be assigned with value at setup()
+    private NotificationAttributes n;
+
+    @BeforeMethod
+    public void setup() throws Exception {
+        for (NotificationAttributes n : typicalNotifications.values()) {
+            notificationsDb.createEntity(n);
+        }
+        n = typicalNotifications.get("notification1");
+    }
+
+    /**
+     * Removes all notifications created by each test.
+     */
+    @AfterMethod
+    public void cleanUp() {
+        List<NotificationAttributes> retrieved = notificationsDb.getAllNotifications();
+        assertNotNull(retrieved);
+
+        retrieved.forEach(n -> notificationsDb.deleteNotification(n.getNotificationId()));
+    }
+
+    @Test
+    public void testGetNotification() throws Exception {
+        ______TS("typical success case");
+        NotificationAttributes retrieved = notificationsDb.getNotification(n.getNotificationId());
+        assertNotNull(retrieved);
+
+        ______TS("expect null for non-existent account");
+        retrieved = notificationsDb.getNotification("invalid_notification_id");
+        assertNull(retrieved);
+
+        ______TS("failure: null parameter");
+        assertThrows(AssertionError.class, () -> notificationsDb.getNotification(null));
+    }
+
+    @Test
+    public void testGetAllNotifications() throws Exception {
+        ______TS("typical success case");
+        List<NotificationAttributes> retrieved = notificationsDb.getAllNotifications();
+
+        // Verifies only the number of returned notifications is correct.
+        // No duplications or wrong data checks are made.
+        assertNotNull(retrieved);
+        assertEquals(typicalNotifications.size(), retrieved.size());
+    }
+
+    @Test
+    public void testGetActiveNotificationsByTargetUser() throws Exception {
+        // Conditions for this API: endTime > now, startTime < now, targetUser == specified target user
+        ______TS("typical success case");
+
+        // This size has to be changed if any changes are made to typicalDataBundle
+        final int expectedReturnSize = 4;
+
+        List<NotificationAttributes> retrieved =
+                notificationsDb.getActiveNotificationsByTargetUser(NotificationTargetUser.STUDENT);
+
+        // Verifies only the number of returned notifications is correct.
+        // No duplications or wrong data checks are made.
+        assertNotNull(retrieved);
+        assertEquals(expectedReturnSize, retrieved.size());
```

Comment:
Checking that the size of the results match up might not be sufficient, as we do not know whether inactive notifications are part of the result or not. 

You can actually do something like this (from `FeedbackSessionsDbTest`):

```
List<FeedbackSessionAttributes> sessions = fsDb.getFeedbackSessionsForCourse("idOfTypicalCourse1");

        String expected =
                dataBundle.feedbackSessions.get("session1InCourse1").toString() + System.lineSeparator()
                + dataBundle.feedbackSessions.get("session2InCourse1").toString() + System.lineSeparator()
                + dataBundle.feedbackSessions.get("empty.session").toString() + System.lineSeparator()
                + dataBundle.feedbackSessions.get("awaiting.session").toString() + System.lineSeparator()
                + dataBundle.feedbackSessions.get("closedSession").toString() + System.lineSeparator()
                + dataBundle.feedbackSessions.get("gracePeriodSession").toString() + System.lineSeparator();

        for (FeedbackSessionAttributes session : sessions) {
            AssertHelper.assertContains(session.toString(), expected);
        }
```